core/src/consensus/tendermint/worker.rs

```diff
@@ -817,47 +878,49 @@ impl Worker {
             self.last_two_thirds_majority =
                 TwoThirdsMajority::from_message(message.on.step.view, message.on.block_hash);
         }
+
+        if vote_step.step == Step::Precommit
+            && self.height == vote_step.height
+            && message.on.block_hash.is_some()
+            && has_enough_aligned_votes
+        {
+            if self.can_move_from_commit_to_propose() {
+                let height = self.height;
+                self.move_to_height(height + 1);
+                self.move_to_step(TendermintState::Propose, is_restoring);
+                return
+            }
+
+            let block_hash = message.on.block_hash.expect("Upper if already checked block hash");
+            if !self.step.is_commit() {
+                self.move_to_step(
+                    TendermintState::Commit {
+                        block_hash,
+                        view: vote_step.view,
+                    },
+                    is_restoring,
+                );
+                return
+            }
+        }
+
```

Comment:
Before this PR, previous rounds' precommit is ignored.
After this PR, previous rounds' precommits are enough received, the node moves to Commit state.